src/audio/dcblock/Kconfig

```diff
@@ -6,3 +6,10 @@ config COMP_DCBLOCK
 	help
 	  Select for DC Blocking Filter component. This component filters out
 	  the DC offset which often originates from a microphone's output.
+
+
+config DCBLOCK_HIFI
+	int "DC Blocking Filter HIFI level (-1 = max)"
+	default -1
```

Comment:
can we depend on ARCH_XTENSA and make this a Kconfig choice (where each choice would depend on ARCH_XTENSA_HIFI_n set by top level PLATFORM Kconfig) ? 

Btw, I think Peter Mitsis will have the HiFi flavour landing in Zephyr soon as part of the HiFi context save/restore work. i.e. we can reuse RTOS supplied Kconfig data.

Whatever methodology we use, it should also scale to x86 and ARM SIMD.

Reply:
where is ARCH_XTENSA and ARCH_XTENSA_HIFI_n definition? is it already exist or added later?

Reply:
also, there is no HIFI option for this module, no HIFI3 and HIFI4 option.

Reply:
> also, there is no HIFI option for this module, no HIFI3 and HIFI4 option.

In this prototype you enter '3' or '4' or any version you like. @lgirdwood already requested a `Kconfig choice`

Please spend more time reading and make sure you understand better before commenting.

Reply:
Just to confirm, I am indeed working on adding HiFi DSP support in Zephyr so that its registers will be saved/restored on relevant context switches.

Reply:
Nice!   Can't resist a preemptive kibitz:  When you do, please do it as a generic hook to save coprocessor state, as Xtensa has some existing FPU context logic that should share the same backend.  And FWIW: lazy-evaluated switching based on the coprocessor enabled exception is really only a nice-to-have in SOF, as the overwhelming majority of processing is done on core-pinned threads that don't preempt each other; storing them all on a switch (but not ISR!) is totally OK.   Similarly will want a hook on interrupt entry to make sure the kernel can't touch the registers incorrectly.  But note eventually this will want userspace support, and syscalls probably **will** want SIMD access, so the syscall entry code will need a spot to do the lazy save.

Reply:
Oh, and remember that the thread struct is uncached, which makes it a poor spot on intel_adsp for saving big blocks like this, you'll want to pass it through arch_xtensa_cached_ptr() and to do your flush/invalidations manually if you want it there, or otherwise find a cached spot for it.

Reply:
I'll shut up now.

Reply:
All - @peter-mitsis will need to define a SOC level Kconfig to indicate the HIFI level on xtensa. i.e. this will say hifi3, hifi4, hifi5 etc as the different hifi ISA add additional register that must be saved/restored.
This same Kconfig can be reused here to indicate the highest HIFI SIMD level for each platform/SOC i.e. some menuconfig options will depend on this Zephyr SOC Kconfig.

Reply:
> All - @peter-mitsis will need to define a SOC level Kconfig to indicate the HIFI level on xtensa.

My attempt to do this:
- https://github.com/zephyrproject-rtos/zephyr/pull/67513

Just prototype and draft but a functional one so please review.

Reply:
Thank you all,  then this open will be resolved soon.
let's wait for Peter's PR to zephyr and got merged.

@peter-mitsis , could you share PR link here once you submit here? or add me into review list for the zephyr PR？

Reply:
#8787 is similar to this but with a Kconfig `choice` instead.